app/src/main/java/org/glucosio/android/activity/MainActivity.java

```diff
@@ -272,6 +274,28 @@ public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
         analytics.reportScreen("Main Activity");
     }
 
+    @Override
+    protected void onResume() {
+        /* Added this method to ensure that new data is shown after being added by other activities
+           example - while viewing HbA1c history, use calculator to add reading.  The new reading
+           wasn't being shown on the history until the user navigated away from the history page
+           and came back, and it should refresh with data IMHO
+         */
+        super.onResume();
+
+        if (viewPager != null && homePagerAdapter != null) {
+            int currentPage = viewPager.getCurrentItem();
+            if (currentPage == 1) {  // assumption - history tab is always position 1
+                HistoryFragment historyFragment = homePagerAdapter.getHistoryFragment();
```

Comment:
You can override onResume in the History fragment and reload data there. However, it is not a super great solution since it will happen also when the user locked the screen, the user opened another app, etc.